src/modules/api.ts

```diff
@@ -317,20 +319,23 @@ function getUnpausedLog(timestamp, Topic) {
   }
 }
 
-export async function release(from, contract) {
+export async function release(from, contract, provider) {
+  const signer = await provider.getSigner()
   if (contract.version === ContractVersion.V1) {
-    return vesting.methods.release().send({ from })
+    return await new Contract(contract.address, vestingAbi, signer).release()
   }
 
-  const releasableAmount = await vesting.methods.getReleasable().call()
+  const ethContract = new Contract(contract.address, periodicTokenVestingAbi, signer)
+  const releasableAmount = await ethContract.getReleasable()
 
-  return vesting.methods.release(from, releasableAmount).send({ from })
+  return await ethContract.release(from, releasableAmount)
 }
 
-export function changeBeneficiary(from, contract) {
+export async function changeBeneficiary(contract, provider) {
+  const signer = await provider.getSigner()
   const { version, address } = contract
 
   return version === ContractVersion.V1
-    ? vesting.methods.changeBeneficiary(address).send({ from })
-    : vesting.methods.setBeneficiary(address).send({ from })
+    ? new Contract(contract.address, vestingAbi, signer).changeBeneficiary(address)
+    : new Contract(contract.address, periodicTokenVestingAbi, signer).setBeneficiary(address)
```

Comment:
This is setting the vesting contract as beneficiary!!!

Reply:
It would lock users out of their vestings 💀

Reply:
🤦‍♂️  Updated!